### PR TITLE
Allow toolbox containers to be created with a custom hostname

### PR DESCRIPTION
**Pull request by buck-ross**
This feature consists of a new optional flag `--hostname HOSTNAME`, which can be used when invoking `toolbox create` to create a new container with a custom hostname.

This is essentially a re-implementation of #210 from scratch, written in golang.

This may also address many of the concerns brought up in the comments of #771, and could potentially assist in the adoption of a new default hostname, since it allows for more flexibility in that regard.

This PR includes:
- [x] A new feature, which is backwards compatible with all existing use-cases of `toolbox create`
- [x] Associated system tests for the new `--hostname` flag
- [x] Associated updates to the `toolbox create` documentation in the manual.

### CODE REVIEW

**Comment by nievesmontero** [comment]
This pull request is solved in https://github.com/containers/toolbox/pull/1134

**Comment by debarshiray** [comment]
> This pull request has been replaced by #1134

Why was this replaced? Was this rebased against main?

File: src/cmd/create.go [hunk 1]
```diff
@@ -95,6 +103,10 @@ func init() {
 }
 
 func create(cmd *cobra.Command, args []string) error {
+	// This regex filters out strings which are not valid hostnames, according to RFC-1123.
+	// Source: https://stackoverflow.com/a/106223
+	var hostnameRegexp = regexp.MustCompile("^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])$")
```
Comment (debarshiray) [inline src/cmd/create.go:108]: Do we really need to have this regular expression directly in the code?  Can't we use some Go package for it?

File: doc/toolbox-create.1.md [hunk 2]
```diff
@@ -85,6 +86,11 @@ Create a toolbox container for a different operating system DISTRO than the
 host. Cannot be used with `--image`. Has to be coupled with `--release` unless
 the selected DISTRO matches the host.
 
+**--hostname** HOSTNAME
+
+Initializes the netowork hostname of the toolbox container to the specified value.
```
Comment (debarshiray) [inline doc/toolbox-create.1.md:91]: Typo alert: *netowork*

File: src/cmd/create.go [hunk 3]
```diff
@@ -115,8 +127,13 @@ func create(cmd *cobra.Command, args []string) error {
 		return errors.New("options --image and --release cannot be used together")
 	}
 
+	if cmd.Flag("hostname").Changed && !hostnameRegexp.MatchString(cmd.Flag("hostname").Value.String()) {
+		return errors.New("invalid hostname")
+	}
+
 	var container string
 	var containerArg string
+	var hostname = cmd.Flag("hostname").Value.String()
```
Comment (debarshiray) [inline src/cmd/create.go:136]: This isn't necessary ...

File: src/cmd/create.go [hunk 4]
```diff
@@ -158,14 +175,14 @@ func create(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if err := createContainer(container, image, release, true); err != nil {
+	if err := createContainer(container, image, release, hostname, true); err != nil {
```
Comment (debarshiray) [inline src/cmd/create.go:178]: ... because you can pass `createFlags.hostname` here.

**Review by debarshiray — COMMENTED** [review]
This is an implementation of https://github.com/containers/toolbox/issues/563  As I mentioned there already, let's first try to improve the default hostname (see https://github.com/containers/toolbox/issues/969).

Customizing the hostname might not always be necessary to distinguish one container from another.  See https://github.com/containers/toolbox/issues/98

Regardless, here are some comments on the specific changes in this pull request.  Maybe they might be of help for future work.

**Pull request closed without merging**